docs/github-api.md

```diff
@@ -26,6 +30,54 @@ module.exports = robot => {
 
 See the [full API docs](https://octokit.github.io/rest.js/) to see all the ways you can interact with GitHub. Some API endpoints are not available on GitHub Apps yet, so check [which ones are available](https://developer.github.com/v3/apps/available-endpoints/) first.
 
+## GraphQL API
+
+> **Heads Up!** GraphQL support in Probot is currently a preview feature and this interface could change in a future releases without notice. [Let us know]() if you have any feedback or ideas to improve GraphQL support.
+
+The [GitHub GraphQL API](https://developer.github.com/v4/) is an alternative
+
+
+Here is an example of the same autoresponder app from above that comments on opened issues, but this time with GraphQL:
+
+```js
+// GraphQL query to get Node id for any resource, which is needed for mutations
+const getResource = `
+  query getResource($url: URI!) {
+    resource(url: $url) {
+      ... on Node {
+        id
+      }
+    }
+  }
+`
+
+// GraphQL query to add a comment
+const addComment = `
+  mutation comment($id: ID!, $body: String!) {
+    addComment(input: {subjectId: $id, body: $body}) {
+      clientMutationId
+    }
+  }
+`
+
+module.exports = robot => {
+  robot.on('issues.opened', async context => {
+    // Get the node id of the issue
+    const { resource } = await context.github.query(getResource, {
+      url: context.payload.issue.html_url
+    })
+
+    // Post a comment on the issue
+    await context.github.query(addComment, {
+      id: resource.id,
+      body: 'Hello World'
+    })
+  })
+}
+```
```

Comment:
I know there was some discussion around `context.github.mutation` as well. Is it possible to limit mutations in the `query` method?

Reply:
Yeah, I think I suggested that initially, but now I'm wondering if there's any value in doing that. I don't think there would be any functional difference between the two methods. I'd be nice if you could leave the `query { ... }` or `mutation { ... }` parts of the query out, but those are required if you use variables.

Would you be opposed to only having one method? Should it be called something besides `query`? I mean, technically a mutation is still a query. `context.github.graphql()` feels weird to me for some reason.

Reply:
fwiw, the `mutation` and `query` keywords in the string can be baked into the relevant method, so even with variables you could still do `context.github.query('{ ... }', { foo: 'bar' })`

Reply:
But the variables need to be declared in the query:

```
  query getResource($url: URI!) { ... }
```